pkg/koordlet/resourceexecutor/resctrl.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourceexecutor
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/util/system"
+)
+
+const ErrResctrlDir = "resctrl path or file not exist"
+const CacheIdIndex = 2
+
+func NewResctrlReader() ResctrlReader {
+	if vendorId, err := system.GetVendorIDByCPUInfo(system.GetCPUInfoPath()); err != nil {
+		klog.V(0).ErrorS(err, "get cpu vendor error, stop start resctrl collector")
+		return &fakeReader{}
+	} else {
+		switch vendorId {
+		case system.INTEL_VENDOR_ID:
+			return NewResctrlRDTReader()
+		case system.AMD_VENDOR_ID:
+			return &ResctrlAMDReader{}
+		default:
```

Comment:
So ARM is unsupported by the collector? I'm still wondering if we can adapt the resctrl interfaces and forget the vendor-specific initialization. We can throw the error if any environment does not support some of the resctrl interfaces. This may be a better backward-compatible solution.

Reply:
Currently, I cannot test on an ARM platform. According to my understanding of this [introduction](https://openanolis.cn/sig/ARM_ARCH_SIG/doc/475502679126360261), ARM's operations rely on ACPI, which contrasts with the x86 method. Any guidance and additional resources on MPAM would be appreciated.

Reply:
We can rely on the resctrl files instead of the vendor specification.

Reply:
> We can depend on the resctrl interfaces instead of the vendor specification. In case any platform does not fully support the resctrl, throw the error.

Fix now.